crates/bevy_pbr/src/meshlet/instance_manager.rs

```diff
@@ -0,0 +1,261 @@
+use super::{meshlet_mesh_manager::MeshletMeshManager, MeshletMesh};
+use crate::{
+    Material, MeshFlags, MeshTransforms, MeshUniform, NotShadowCaster, NotShadowReceiver,
+    PreviousGlobalTransform, RenderMaterialInstances,
+};
+use bevy_asset::{AssetEvent, AssetServer, Assets, Handle, UntypedAssetId};
+use bevy_ecs::{
+    entity::{Entities, Entity, EntityHashMap},
+    event::EventReader,
+    query::Has,
+    system::{Local, Query, Res, ResMut, Resource, SystemState},
+};
+use bevy_render::{render_resource::StorageBuffer, view::RenderLayers, MainWorld};
+use bevy_transform::components::GlobalTransform;
+use bevy_utils::{HashMap, HashSet};
+use std::ops::{DerefMut, Range};
+
+/// Manages data for each entity with a [`MeshletMesh`].
+#[derive(Resource)]
+pub struct InstanceManager {
```

Comment:
Maybe it's fine with the name space, but that feels a bit generic? Why not something like `VirtualGeometryMeshManager`? To be clear, this isn't a blocker or anything, I'm just curious.

Reply:
Eh I didn't want to get hung up on naming. We can change it later.